src/HDF5.jl

```diff
@@ -656,15 +657,16 @@ Open or create an HDF5 file where `mode` is one of:
 Pass `swmr=true` to enable (Single Writer Multiple Reader) SWMR write access for "w" and
 "r+", or SWMR read access for "r".
 """
-function h5open(filename::AbstractString, mode::AbstractString="r", pv...; swmr=false)
+function h5open(filename::AbstractString, mode::AbstractString="r", pv...; swmr=false, fclose_degree = H5F_CLOSE_STRONG, auto_fclose = true)
     checkprops(pv...)
     # pv is interpreted as pairs of arguments
     # the first of a pair is a key of hdf5_prop_get_set
     # the second of a pair is a property value
     fapl = p_create(H5P_FILE_ACCESS, true, pv...) # file access property list
     # With garbage collection, the other modes don't make sense
     # (Set this first, so that the user-passed properties can overwrite this.)
-    fapl["fclose_degree"] = H5F_CLOSE_STRONG
+    fclose_degree != H5F_CLOSE_STRONG && @eval GC.gc(true)
```

Comment:
why this?

Reply:
Sorry, I meant why the GC.gc on this line

Reply:
https://github.com/JuliaIO/HDF5.jl/issues/620#issuecomment-619417143 
So maybe we are missing something similar here causing the prevention of reading the virutal dataset?

I just don't understand the necessity of this `GC.gc(true)` which we should try to avoid.

Reply:
#621 didn't solve this problem